libPhoneNumber/NBPhoneNumberUtil.m

```diff
@@ -344,17 +345,9 @@ - (BOOL)isAllDigits:(NSString *)sourceString {
   return r.location == NSNotFound;
 }
 
-- (BOOL)isNumeric:(NSString *)sourceString {
-  NSScanner *sc = [NSScanner scannerWithString:sourceString];
-  if ([sc scanFloat:NULL]) {
-    return [sc isAtEnd];
-  }
-  return NO;
-}
-
 - (BOOL)isNaN:(NSString *)sourceString {
-  if ([self isNumeric:sourceString]) return NO;
-  return YES;
+  // Return YES if the sourceString doesn't have any characters that can be represented as a Float.
```

Comment:
Note that 2.4E15 is a perfectly acceptable float... Also, if you rewrote this as a function instead of a method you would get a smaller binary and faster calling. There is no reason for it to be a method of this class.

Reply:
Converted to a static function.

So it looks like this function is only used to filter out non-geographical calling code, which can only be an integer, so I changed the code to reflect that.